File: solutions/ruby/nth-prime/1/nth_prime.rb

module Prime
  def self.prime?(number)
    (2..(number**0.5)).find{|i| number % i == 0}.nil?
  end

  def self.generator
    Enumerator.new do |enum|
      number = 2

      loop do
        if prime?(number)
          enum.yield number
        end
        number += 1
      end
    end
  end

  def self.nth(n)
    raise ArgumentError.new() if n <= 0

    prime_generator = generator
    (0..n-1).map{prime_generator.next}.last
  end
end

File: solutions/python/nth-prime/1/nth_prime.py
def prime_generator():
    number = 2

    while True:
        for i in range(2, int(number**.5 + 1)):
            if number % i == 0:
                break
        else:
            yield number
        number += 1


def prime(number):
    if number == 0:
        raise ValueError("there is no zeroth prime")

    gen = prime_generator()
    for _ in range(number):
        prime = next(gen)

    return prime
